`argo template lint` command panic

## Summary 

`argo template lint` command panic

## Diagnostics


```
# argo template lint
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/argoproj/argo/cmd/argo/commands/template.ServerSideLint(0x3d36df0, 0x0, 0x0, 0x1, 0x2beb810, 0x5)
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/commands/template/lint.go:39 +0x686
github.com/argoproj/argo/cmd/argo/commands/template.NewLintCommand.func1(0xc0006f7400, 0x3d36df0, 0x0, 0x0)
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/commands/template/lint.go:27 +0x54
github.com/spf13/cobra.(*Command).execute(0xc0006f7400, 0x3d36df0, 0x0, 0x0, 0xc0006f7400, 0x3d36df0)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xc0006ec780, 0xc0000f0750, 0xc0009fbf50, 0x40576f)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/acollins8/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/Users/acollins8/go/src/github.com/argoproj/argo/cmd/argo/main.go:17 +0x2b
```




What version of Argo Workflows are you running? 

```
# argo version
argo: v2.11.5
  BuildDate: 2020-10-15T22:26:34Z
  GitCommit: 076bf89c4658adbd3b96050599f81424d1b08d6e
  GitTreeState: clean
  GitTag: v2.11.5
  GoVersion: go1.13.15
  Compiler: gc
  Platform: linux/amd64
```


---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/cmd/argo/commands/template/lint.go
+++ b/cmd/argo/commands/template/lint.go
@@ -24,6 +24,10 @@ func NewLintCommand() *cobra.Command {
 		Use:   "lint (DIRECTORY | FILE1 FILE2 FILE3...)",
 		Short: "validate a file or directory of workflow template manifests",
 		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				cmd.HelpFunc()(cmd, args)
+				os.Exit(1)
+			}
 			err := ServerSideLint(args, strict)
 			if err != nil {
 				log.Fatal(err)

--- a/test/e2e/cli_test.go
+++ b/test/e2e/cli_test.go
@@ -725,6 +725,14 @@ func (s *CLIWithServerSuite) TestWorkflowWatch() {
 }
 
 func (s *CLISuite) TestTemplate() {
+	s.Run("LintWithoutArgs", func() {
+		s.Given().RunCli([]string{"template", "lint"}, func(t *testing.T, output string, err error) {
+			if assert.Error(t, err) {
+				assert.Contains(t, output, "Usage:")
+			}
+		})
+	})
+
 	s.Run("Lint", func() {
 		s.Given().RunCli([]string{"template", "lint", "smoke/workflow-template-whalesay-template.yaml"}, func(t *testing.T, output string, err error) {
 			if assert.NoError(t, err) {

